site/content/docs/custom-layout.md

```diff
@@ -0,0 +1,149 @@
+---
+title: How to create custom page types and layouts
+description: You may have many different types of notes that you want to display in different ways on your Flowershow website. In this tutorial we will cover how to add a new `Recipe` document type for our cooking blog as well as a custom `RecipeLayout` for that specific document type.
+type: Blog
+created: 2023-03-02
+authors: [philippe-du-preez]
+---
+
+You may have many different types of notes that you want to display in different ways on your Flowershow website. In this tutorial we will cover how to add a new `Recipe` document type for our cooking blog as well as a custom `RecipeLayout` for that specific document type. Then we will create a home page for listing all the pages of this type.
+
+Let's start with creating a new `Recipe` document type.
+
+## Create a custom Contentlayer document type
+
+In order to create our new `Recipe` document type, we need to go inside `.flowershow/contentlayer.config.ts` and add instructions to tell Contentlayer how to parse:
+
+- `name`: namespace
+- `filePathPattern`: input files (Files within this path do not need to specify their `Type` to `Recipe`)
+- `fields`: meta data fields
+- `computedFields`: derived meta data fields
```

Comment:
I don't really understand this part 😅 I'd rather expect some intro, saying what metadata and other data will be needed in all of our recipe pages, e.g. prep time, cook time, easy/medium/hard, for how many serves it is, maybe cost, image, and obviously ingredients. Then I'd show how to create a document of this type in contentlayer config.